quetz/main.py

```diff
@@ -1798,14 +1801,24 @@ def serve_path(
         except ValueError:
             pass
 
-    if is_package_request and channel.mirror_channel_url:
+    if is_package_request and channel.mirror_channel_urls:
         # if we exclude the package from syncing, redirect to original URL
+        # use the first mirror url entry for the redirect
         channel_proxylist = json.loads(channel.channel_metadata).get('proxylist', [])
         if channel_proxylist and package_name and package_name in channel_proxylist:
-            return RedirectResponse(f"{channel.mirror_channel_url}/{path}")
-
-    if channel.mirror_channel_url and channel.mirror_mode == "proxy":
-        repository = RemoteRepository(channel.mirror_channel_url, session)
+            # proxying packages only works for the first url in the list
+            redirect_url = channel.mirror_channel_urls[0]
+            return RedirectResponse(f"{redirect_url}/{path}")
+
+    # note: you can only proxy, when you have exactly one remote channel in your config
+    if channel.mirror_channel_urls and channel.mirror_mode == "proxy":
+        proxy_url = channel.mirror_channel_urls[0]
+        if len(channel.mirror_channel_urls) > 1:
+            logger.warning(
```

Comment:
I think generally warnings about user-input in a webserver are a bit ineffective because the person who is doing something wrong never gets to see the warning. Should this just be a failure?

Reply:
Let's go with the route to directly disable this on channel creation time

Reply:
It has been done